### PR TITLE
Make Fortran AMO type tables consistent/correct

### PR DESCRIPTION
**Pull request by davidozog**
These are just minor issues with the Fortran descriptions, but might as well fix them while we're at it.

### CODE REVIEW

**Comment by nspark** [comment]
For reference, the original text that this PR modifies goes back to OpenSHMEM 1.1, but these changes correct both inconsistencies and inaccuracies (e.g., using the wrong name for a variable) that should be fixed.

File: content/shmem_atomic_compare_swap.tex [hunk 1]
```diff
@@ -46,8 +46,7 @@
     atomic operation.
 }
 \apidesctable{
-    The  \VAR{dest} and \VAR{value} data objects must conform to certain typing
-    constraints, which are as follows:
+    When using \Fortran, \VAR{dest} and \VAR{value} must be of the following type:
```
Comment (nspark) [inline content/shmem_atomic_compare_swap.tex]: Should this one also include `\VAR{cond}`?

Comment (nspark) [reply content/shmem_atomic_compare_swap.tex]: Poke @davidozog

Comment (davidozog) [reply content/shmem_atomic_compare_swap.tex]: Ah yes.  Sorry, I'm on vacation this week.  I've made the change.

Comment (nspark) [reply content/shmem_atomic_compare_swap.tex]: Thanks! Sorry to bug you on vacation!

**Pull request merged**